app/controlplane/pkg/data/workflowrun.go

```diff
@@ -240,6 +277,17 @@ func entWrToBizWr(ctx context.Context, wr *ent.WorkflowRun) *biz.WorkflowRun {
 		r.Workflow = w
 	}
 
+	// Load version preloaded or otherwise query it
```

Comment:
This will potentially introduce a N+1 problem in the data layer (when listing workflow runs, for example, we will make this query multiple times). I'd suggest preload it with a inner join.

Reply:
it is, in practice, preloaded (see code above).

![image](https://github.com/user-attachments/assets/b5026f2a-2377-4925-9c4a-aa2fc0cb6f9a)

This check is to make sure the code works regardless of the preloading status